viewing_party/party.py

```diff
@@ -1,23 +1,192 @@
-    pass
+    Keyword arguments:
+    title -- a string movie title
+    genre -- a string genre name
+    rating -- a float value
+    """
+    if title and genre and rating:
+        movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data: dict, movie: dict):
+    """ Add a movie dictionary to the "watched" key in user_data.
+    Return user_data.
+
+    Keyword arguments:
+    user_data -- a dictionary with a "watched" key with a list value
+    movie -- a dictionary with movie info
+    """
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data: dict, movie: dict):
+    """ Add a movie dictionary to the "watchlist" key in user_data.
+    Return user_data.
+
+    Keyword arguments:
+    user_data -- a dictionary with a "watchlist" key with a list value
+    movie -- a diictionary with movie info
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data: dict, title: str):
+    """ Move a movie in user_data from "watchlist" to "watched".
+    Return user_data.
+
+    Keyword arguments:
+    user_data -- a dictionary with "watchlist" and "watched" keys
+    title -- the title of the movie being moved
+    """
+    counter = 0
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            movie_watched = user_data["watchlist"].pop(counter)
+            user_data["watched"].append(movie_watched)
+            return user_data
+        counter += 1
+    else:
+        return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data: dict):
+    """ Return the average rating for "watched" movies in user_data.
+    
+    Keyword arguments:
+    user_data -- a dictionary with a "watched" key with a list value
+    """
+    num_movies_watched = len(user_data["watched"])
+    all_ratings = 0.0
+
+    if num_movies_watched == 0:
+        return all_ratings
+    else:
+        for movie in user_data["watched"]:
+            all_ratings += movie["rating"]
+        average_rating = all_ratings / num_movies_watched
+        return average_rating
+
+def get_most_watched_genre(user_data: dict):
+    """ Return the most-watched genre in user_data.
+    
+    Keyword arguments:
+    user_data -- a dictionary with a "watched" key with a list value
+    """
+    all_genres = {}
+    most_watched_genre = None
 
+    if len(user_data["watched"]) == 0:
+        return most_watched_genre
+    else:
+        for movie in user_data["watched"]:
+            if movie["genre"] in all_genres:
+                all_genres[movie["genre"]] += 1
+            else:
+                all_genres[movie["genre"]] = 1
+        most_watched_genre = max(all_genres, key=all_genres.get)

```

Comment:
Nice use of max here. I find it useful to also practice writing your own implementation of finding a max value from a dictionary. You may find in an interview if you use built in functions like max that an interviewer may ask you how the function works under the hood. So knowing how to write it on your own can help you better understand the function.